crates/red_knot_python_semantic/src/types.rs

```diff
@@ -4312,6 +4330,18 @@ pub struct BoundMethodType<'db> {
     self_instance: Type<'db>,
 }
 
+impl<'db> BoundMethodType<'db> {
+    pub(crate) fn into_callable_type(self, db: &'db dyn Db) -> Type<'db> {
+        let signature = self.function(db).signature(db).clone();
```

Comment:
Is the `clone` call here necessary, considering that we already clone each parameter on the line below